### PR TITLE
Rename minReplicationFactor to writeConcern

### PR DESCRIPTION
**Pull request by graetzer**
### Scope & Purpose

Rename minRepliationFactor to avoid confusion with `--cluster.`

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests

### Documentation

- [x] Added a *Changelog Entry*
- [x] Added entry to *Release Notes* 
- [x] Added a new section in the *http API* : https://github.com/arangodb/docs/pull/249

http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/9242/
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7455/

### CODE REVIEW

File: arangod/VocBase/Methods/Collections.cpp [hunk 1]
```diff
@@ -602,10 +602,12 @@ Result Collections::properties(Context& ctxt, VPackBuilder& builder) {
 }
 
 Result Collections::updateProperties(LogicalCollection& collection,
-                                     velocypack::Slice const& props, bool partialUpdate) {
+                                     velocypack::Slice const& props) {
+  const bool partialUpdate = false; // always a full update
```
Comment (jsteemann) [inline arangod/VocBase/Methods/Collections.cpp]: it looks like `partialUpdate` is passed as a parameter to `LogicalCollection::properties()`, but there it does not seem to be used at all. So can we remove this variable and remove it from the signature of `LogicalCollection::properties()`?

File: arangosh/Restore/RestoreFeature.cpp [hunk 2]
```diff
@@ -308,9 +308,10 @@ arangodb::Result tryCreateDatabase(arangodb::application_features::ApplicationSe
     if (properties.isObject()) {
       ObjectBuilder guard(&builder, "options");
       for(auto const& key : std::vector<std::string>{
-        arangodb::StaticStrings::MinReplicationFactor,
+        arangodb::StaticStrings::MinReplicationFactor, // deprecated
```
Comment (jsteemann) [inline arangosh/Restore/RestoreFeature.cpp]: I think we can remove `minReplicationFactor` here, as it is a new API in 3.6. And 3.5 never used `minReplicationFactor` for databases.

File: arangod/VocBase/Methods/Collections.cpp (hunk 1 above)
Comment (graetzer) [reply arangod/VocBase/Methods/Collections.cpp]: it is used in Views which inherit this method from logicaldatasource

Comment (jsteemann) [reply arangod/VocBase/Methods/Collections.cpp]: oh, ok.

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/Documentation/job/arangodb-ANY-examples/461/

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7474/

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**